packages/@ember/-internals/metal/lib/is_empty.ts

```diff
@@ -53,17 +53,17 @@ export default function isEmpty(obj: any): boolean {
       return !size;
     }
   }
-
-  if (typeof obj.length === 'number' && objectType !== 'function') {
-    return !obj.length;
-  }
-
+  
   if (objectType === 'object') {
     let length = get(obj, 'length');
     if (typeof length === 'number') {
       return !length;
     }
   }
+  
+  if (typeof obj.length === 'number' && objectType !== 'function') {
```

Comment:
@rwjblue you had a comment about this check, but removing it made tests fail so I didn't touch it.

Reply:
I just don't grok how this can ever be hit. The check **just above** here is checking all `"object"`s that have a numeric `length` property.

Reply:
@rwjblue it has different `objectType` (object vs function)

Reply:
looks like it's true for string case `typeof 'foo'.length === 'number' && typeof 'foo' !== 'function'`